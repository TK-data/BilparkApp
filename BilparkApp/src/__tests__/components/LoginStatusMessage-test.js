import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LoginStatusMessage from '../../components/LoginStatusMessage';

const initialState = {
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    isLoggedIn: true,
    user: '{"UserID":111,"Email":"awkdnawpidn@aiwfaf.no","CompanyID":7,"CarID":82,"Address":"Oiabfoaiwbf","Fname":"Obafiobafiob","Lname":"Oabfoaib","FuelDay":0,"FuelTime":"12-00","FuelNotification":false,"createdAt":"2018-05-20T11:30:10.000Z","updatedAt":"2018-05-20T11:30:21.000Z"}',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing LoginStatusMessage', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <LoginStatusMessage />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
