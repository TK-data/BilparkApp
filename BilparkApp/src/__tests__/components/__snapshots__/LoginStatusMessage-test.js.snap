// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginStatusMessage Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginStatusMessage
    dispatch={[Function]}
    isLoggedIn={true}
    user="{\\"UserID\\":111,\\"Email\\":\\"awkdnawpidn@aiwfaf.no\\",\\"CompanyID\\":7,\\"CarID\\":82,\\"Address\\":\\"Oiabfoaiwbf\\",\\"Fname\\":\\"Obafiobafiob\\",\\"Lname\\":\\"Oabfoaib\\",\\"FuelDay\\":0,\\"FuelTime\\":\\"12-00\\",\\"FuelNotification\\":false,\\"createdAt\\":\\"2018-05-20T11:30:10.000Z\\",\\"updatedAt\\":\\"2018-05-20T11:30:21.000Z\\"}"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Du er logget inn som!
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        />,
        <Button
          onPress={[Function]}
          title="Profil"
        />,
        <Button
          onPress={[Function]}
          title="Register bil"
        />,
        <Button
          onPress={[Function]}
          title="Planlegging av tanking"
        />,
        <Button
          onPress={[Function]}
          title="Meny"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Du er logget inn som!",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Du er logget inn som!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "Profil",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "Register bil",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "Planlegging av tanking",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "Meny",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Du er logget inn som!
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}
          />,
          <Button
            onPress={[Function]}
            title="Profil"
          />,
          <Button
            onPress={[Function]}
            title="Register bil"
          />,
          <Button
            onPress={[Function]}
            title="Planlegging av tanking"
          />,
          <Button
            onPress={[Function]}
            title="Meny"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Du er logget inn som!",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Du er logget inn som!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Profil",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Register bil",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Planlegging av tanking",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Meny",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
